### PR TITLE
Feature new disease anthrax

### PR DESCRIPTION
**Pull request by faatihi**
#833

### CODE REVIEW

**Comment by MartinWahnschaffeSymeda** [comment]
@faatihi Is this ready to be reviewed?

**Comment by faatihi** [comment]
Yes, it's ready

**Comment by MartinWahnschaffeSymeda** [comment]
@faatihi Do you have an estimate on when you can provide this with the new fields that Bernard posted in #833?

**Comment by faatihi** [comment]
I should be done with it by the end of this Thursday.

**Comment by faatihi** [comment]
Hi, has anyone come across this error?

image

After updating to 1.30.0, I get this error when I try going to case details

**Comment by MateStrysewskeSym** [comment]
@faatihi Have you tried re-deploying the server or restarting your development environment? NoClassDefFroundErrors are typically associated with unsuccessful deployment - though it could also be a different issue.

**Comment by faatihi** [comment]
> Hi, has anyone come across this error?
> 
> java.lang.NoClassDefFoundError: de/symeda/auditlog/api/value/format/UtilDateFormatter$1
> 
> image
> 
> After updating to 1.30.0, I get this error when I try going to case details

Never mind. The error is gone.

**Comment by faatihi** [comment]
@MateStrysewskeSym I've finished implementing all requirements but having tough time running the builds for testing.
If you wouldn't mind, you may review this request while I figure a way to run it locally.

File: sormas-api/src/main/java/de/symeda/sormas/api/symptoms/SymptomsDto.java [hunk 1]
```diff
@@ -481,9 +483,13 @@ public static SymptomsDto build() {
...
 	@Diseases({Disease.EVD,Disease.LASSA,Disease.YELLOW_FEVER,Disease.DENGUE,Disease.PLAGUE,Disease.UNSPECIFIED_VHF,Disease.UNDEFINED,Disease.OTHER})
 	private SymptomState unexplainedBleeding;
-	@Diseases({Disease.EVD,Disease.LASSA,Disease.NEW_INFLUENCA,Disease.CSM,Disease.CHOLERA,Disease.MEASLES,Disease.YELLOW_FEVER,Disease.DENGUE,Disease.MONKEYPOX,Disease.PLAGUE,Disease.UNSPECIFIED_VHF,Disease.UNDEFINED,Disease.OTHER})
+	@Diseases({Disease.EVD,Disease.LASSA,Disease.NEW_INFLUENCA,Disease.CSM,Disease.CHOLERA,Disease.MEASLES,Disease.YELLOW_FEVER,Disease.DENGUE,Disease.MONKEYPOX,Disease.PLAGUE,Disease.ANTHRAX,Disease.UNSPECIFIED_VHF,Disease.UNDEFINED,Disease.OTHER})
 	@Outbreaks
 	private SymptomState vomiting;
+	@Diseases({Disease.ANTHRAX})
+	private SymptomState convulsion;
+	@Diseases({Disease.ANTHRAX})
+	private SymptomState skinLesion; /* oedema */
```
Comment (MateStrysewskeSym) [inline sormas-api/src/main/java/de/symeda/sormas/api/symptoms/SymptomsDto.java]: Take this symptom out; I've talked to Bernard and he said that we should only use one of the lesion symptoms, i.e., the one that's already existing

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 2]
```diff
@@ -1252,6 +1252,10 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 							"requestedOtherPathogenTests, requestedOtherAdditionalTests, 'EXTERNAL', pathogenTestResult, " +
 							"changeDate, creationDate, id, lastOpenedDate, localChangeDate, modified, snapshot, uuid FROM tmp_samples;");
 					getDao(Sample.class).executeRaw("DROP TABLE tmp_samples;");
+				case 168:
+					currentVersion = 168;
+					getDao(Symptoms.class).executeRaw("ALTER TABLE symptoms ADD COLUMN convulsion varchar(255);");
+					getDao(Symptoms.class).executeRaw("ALTER TABLE symptoms ADD COLUMN skinLesion varchar(255);");
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: Please make sure that you always update the `currentVersion` variable at the top of the class whenever you add new lines here; it always has to be the last value assigned in this method + 1 (so in this case, 169); also, please make sure to use the correct number here after you've merged the development branch into the pull request

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/symptoms/Symptoms.java [hunk 3]
```diff
@@ -1140,6 +1144,11 @@ public void setTremor(SymptomState tremor) {
         this.tremor = tremor;
     }
 
+    public SymptomState getConvulsion() { return convulsion; }
+    public void setConvulsion(SymptomState convulsion) { this.convulsion = convulsion; }
+    public SymptomState getSkinLesion() { return skinLesion; }
+    public void setSkinLesion(SymptomState skinLesion) { this.skinLesion = skinLesion; }
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/symptoms/Symptoms.java]: Please use line breaks

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java (hunk 2 above)
Comment (faatihi) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: "also, please make sure to use the correct number here after you've merged the development branch into the pull request"

I dont understand this part

Comment (MateStrysewskeSym) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: It's just something we noted in one of the last pull requests; you (or Joshua) used 168 (I believe) there even though the version before was 165, i.e. you skipped two versions.

Comment (faatihi) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: Okay, well note.

Comment (faatihi) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: "Please make sure that you always update the `currentVersion` variable at the top of the class"

you mean `DATABASE_VERSION`?

Comment (MateStrysewskeSym) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java]: @faatihi Yes, absolutely correct!

**Pull request merged**